File: src/api/get_files.php
<?php

require_once __DIR__.'/tools/init.php';

requestShouldBe('get');

// allow authorized users only
setRestrictedAccess();

$files = dbQuery('
    SELECT f.id, f.name, LENGTH(fv.content) as size
    FROM files f
        LEFT JOIN files_versions fv ON fv.file_id = f.id
    WHERE f.user_id = :user_id
    GROUP BY f.id
    ORDER BY f.date_created ASC, fv.date_created DESC
', ['user_id' => getUserId()]);

_success(['files' => $files]);

File: src/aws/lambda/grinotes_lambda_functions/lambda/get_files.py
import sys
import json
import logging
from tools import db
import pymysql

def lambda_handler(event, context):
    """
    This function fetches the files list of a user
    params:
        event.fbid
    """

    # receive parameters
    fbid = event['fbid']

    filesList = db.dbQuery(
        'SELECT f.id, f.name, LENGTH(fv.content) as size '+
        'FROM files f ' +
            'LEFT JOIN files_versions fv ON fv.file_id = f.id '+
        'WHERE f.user_id = %s '+
        'GROUP BY f.id '+
        'ORDER BY f.date_created ASC, fv.date_created DESC',
        fbid
    )

    return {'result':'ok', 'files':filesList}
